src/Apps/Order/Routes/Shipping2/Hooks/useBackToFulfillmentDetails.tsx

```diff
@@ -0,0 +1,64 @@
+import { usePrevious } from "@artsy/palette"
+import { useShippingContext } from "Apps/Order/Routes/Shipping2/Hooks/useShippingContext"
+import { matchAddressFields } from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
+import { extractNodes } from "Utils/extractNodes"
+import { Shipping2_me$data } from "__generated__/Shipping2_me.graphql"
+import { useEffect } from "react"
+
+/**
+ * Go back to fulfillment details stage if the user edits the address or
+ * deletes a saved address.
+ */
+export const useBackToFullfillmentDetails = (me: Shipping2_me$data) => {
```

Comment:
Moves behavior out of the `index` parent container and into a self-describing hook. Now we've got a single unit that is only concerned with what it needs.

Reply:
@erikdstock - At some point we can start moving a lot of the tests that happen in `Shipping2/index.tsx` into individual unit tests on our new hooks, and then from there, lean more heavily on integrity. 

Right now there's an enormous amount of integration-like testing in that one file which is very hard to understand and augment, which could be offloaded to an actual integration testing library in a much simpler fashion. If we test these key hooks and behavior in the context, we can maintain safety while reducing complexity in the test suite.

Reply:
yep, I actually just merged some of that that in #13293